sw/device/lib/base/log.c

```diff
@@ -63,42 +57,42 @@ void base_log_internal_core(log_severity_t severity, const char *file_name,
   // nothing was printed for some time.
   static uint16_t global_log_counter = 0;
 
-  base_printf("%s%5d %s:%d] ", stringify_severity(severity), global_log_counter,
-              base_name, line);
+  base_printf("%s%5d %s:%d] ", stringify_severity(log.severity),
+              global_log_counter, base_name, log.line);
   ++global_log_counter;
 
   va_list args;
-  va_start(args, format);
-  base_vprintf(format, args);
+  va_start(args, log);
+  base_vprintf(log.format, args);
   va_end(args);
 
   base_printf("\r\n");
 }
 
-/**
- * Logs `format` and the values that following in an efficient, DV-testbench
- * specific way.
- *
- * @param log the log_fields_t struct that holds the log fields.
- * @param nargs the number of arguments passed to the format string.
- * @param ... format parameters matching the format string.
- */
-
 /**
  * Indicates the fixed location in RAM for SW logging for DV.
  * TODO: Figure aout a better place to put this.
  */
 static const uintptr_t kSwLogDvAddr = 0x1000fffc;
 
-void base_log_internal_dv(const log_fields_t *log, int nargs, ...) {
-  mmio_region_t sw_log_dv_addr = mmio_region_from_addr(kSwLogDvAddr);
-  mmio_region_write32(sw_log_dv_addr, 0x0, (uintptr_t)log);
+/**
+ * Logs `log and the values that follow in an efficient, DV-testbench
+ * specific way.
+ *
+ * @param log a pointer to log data to log. Note that this pointer is likely to
+ *        be invalid at runtime, since the pointed-to data will have been
+ *        stripped from the binary.
```

Comment:
```suggestion
 *        stripped from the binary. As such, it is immaterial to the rest of the software.
          It is only used by the testbench that is monitoring the address to which its
          value is being written to. 
```

Reply:
I don't think this is quite the meaning I'm going for here; the key thing is that this pointer cannot be dereferenced, since the function cannot know if it was valid or not. The point is to make minimal assurances about the pointer.

Reply:
"Likely to be invalid" IMO gives the wrong message. It IS invalid and it cannot be dereferenced. The last part explains why. In any case, I know how it works so I really dont care much about how clear we want to be with this explanation. Feel free to leave this as it is. I'd however recommend the first two changes (remove "likely to be").